src/js/file-manager.js

```diff
@@ -163,4 +171,68 @@ fileManager.isTooLarge = () => { return false; };
  */
 fileManager.getMaxSizeReadable = () => { return `${5}MB`; };
 
+/**
+ * @function getMaxImageWidth
+ *
+ * @description Replace with function that determines maximum image width.
+ *
+ * @return {number} maximum image width
+ */
+fileManager.getMaxImageWidth = () => { return 1024; };
+
+/**
+ * @function isImageFile
+ *
+ * @description Check whether current file is image by reading its type.
+ *
+ * @return {boolean} whether file is image
+ */
+fileManager.isImageFile = file => { return file && file.type.split( '/' )[ 0 ] === 'image'; };
+
+/**
+ * @function resizeImage
+ *
+ * @param {File} file - image file to be resized
+ * @param {number} maxWidth - maximum width of resized image
+ * @param {number} maxHeight - maximum height of resized image
+ *
+ * @return {Promise<Blob>} promise of resized image blob
+ */
+fileManager.resizeImage = ( file, maxWidth, maxHeight ) => {
+    return new Promise( ( resolve, reject ) => {
```

Comment:
I think the method should be called from the widget.

Reply:
Method now called from `Filepicker`.
Little question. What should I do when resizeImage is successfull?

Reply:
Good question. I haven't figured this out myself. Unfortunately, it is not possible to dynamically replace the file in a `<input type="file">`.


I think we have make a tweak to `getCurrentFiles` in the file-manager, and in the filepicker add an attribute to the file input if it was resized. So if that attribute exists getCurrentFiles will not use `this.files[0]` to get the file but gets the resized blob instead (not sure how though...). We do something a little similar with the drawing widget/canvas. 

It would also be good to use the resized blob for the preview, so users don't get a false sense of picture quality to be uploaded, if the quality was actually visibly reduced a lot.

Reply:
Now image preview and getCurrentFiles using resized image file.